src/coins.cpp

```diff
@@ -250,7 +250,10 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn
 
 bool CCoinsViewCache::Flush() {
     bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase=*/ true);
-    cacheCoins.clear();
+    if (cacheCoins.size() != 0) {
+        /* BatchWrite must erase all cacheCoins elements when erase=true. */
```

Comment:
A few minor suggestions.

```diff
 
 bool CCoinsViewCache::Flush() {
-    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase=*/ true);
-    if (cacheCoins.size() != 0) {
-        /* BatchWrite must erase all cacheCoins elements when erase=true. */
+    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase=*/true);
+    if (!cacheCoins.empty()) {
+        // BatchWrite must erase all cacheCoins elements when erase=true.
         throw std::logic_error("Not all cached coins were erased");
     }
     cachedCoinsUsage = 0;
@@ -260,7 +260,7 @@ bool CCoinsViewCache::Flush() {
 
 bool CCoinsViewCache::Sync()
 {
-    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase=*/ false);
+    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase=*/false);
     // Instead of clearing `cacheCoins` as we would in Flush(), just clear the
```